tests/route_middleware.rs

```diff
@@ -128,3 +128,18 @@ async fn subroute_not_nested() -> tide::Result<()> {
     assert_eq!(res["x-child"], "child");
     Ok(())
 }
+
+#[async_std::test]
+async fn app_middleware_reset() -> tide::Result<()> {
+    let mut app = tide::new();
+    app.with(TestMiddleware::with_header_name("X-Root", "root"));
+    app.reset_middleware();
+    app.at("/foo")
+        .with(TestMiddleware::with_header_name("X-Foo", "foo"))
+        .get(echo_path);
+
+    let res = app.get("/foo").await?;
+    assert!(res.header("X-Root").is_none());
+    assert_eq!(res["X-Foo"], "foo");
+    Ok(())
+}
```

Comment:
A test should be added to ensure that previously defined routes on the `app` (i.e. those before line 136) still use the initially set middleware.